official/transformer/utils/schedule.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
```

Comment:
Much of this logic seems very specific to Transformer. Why in utils? The TPU logic should be separated out, to make things more clear.

Reply:
It's in `official/transformer/utils`, not `official/utils`. That should be fine, no?